build.gradle

```diff
@@ -18,38 +18,42 @@
  */
 
 plugins {
-	id 'com.gradle.plugin-publish' version '0.14.0'
-	id 'java-gradle-plugin'
+    id 'checkstyle'
+    id 'java-gradle-plugin'
 }
 
-apply plugin: 'maven-publish'
-apply plugin: 'checkstyle'
-
-group = 'org.jruyi.gradle'
-archivesBaseName = 'thrift-gradle-plugin'
-version = '0.4.2'
-description = 'A gradle plugin for compiling Thrift IDL files using thrift compiler'
-
 repositories {
-	mavenCentral()
+    mavenCentral()
 }
 
 dependencies {
-	testImplementation "org.assertj:assertj-core:3.24.2"
-	testImplementation "org.junit.jupiter:junit-jupiter:5.9.3"
-	testImplementation "org.mockito:mockito-junit-jupiter:5.4.0"
-	testImplementation "org.mockito.kotlin:mockito-kotlin:5.0.0"
+    testImplementation "org.assertj:assertj-core:3.24.2"
+    testImplementation "org.junit.jupiter:junit-jupiter:5.9.3"
+    testImplementation "org.mockito:mockito-junit-jupiter:5.4.0"
+    testImplementation "org.mockito.kotlin:mockito-kotlin:5.0.0"
+}
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(8)
```

Comment:
If `languageVersion` is set to 8, don't we need to install Java 8 in CI builds?

Reply:
Gradle will check local and download it automatically for us if it can find the appropriate version from https://api.adoptium.net/q/swagger-ui/. So for mac+aarch64 we may need to download manually, but other version can just write like this and let it download for us(e.g. Github Action is downloading automatically)

Reply:
Yeah, Gradle automatically downloads it but it should be inefficient to download Java 17 if we only use Java 8 to build and test projects.

Reply:
I realized that we use Java 17 for checkstyle. I prefer to build the code JDK 17 and set `-release` to 8 to produce byte codes for Java 8.

Reply:
Not familiar with `languageVersion = JavaLanguageVersion.of(8)`. But I think it will make our gradle compile/build the code with Java8 (result will be something like --release 8?). So we can using any version of JDK which gradle runtime supports to run the gradle project and build the specific version result? 

I tried the build tasks, and by default current setting will generate byte code 52. And if we use newer API, the compile task will complain too.

Reply:
Related to another comment.

We have two options? 

* Setting like below only and run gradle with 8, 11, 17 JDK
```
tasks.withType(JavaCompile).configureEach {
    sourceCompatibility = '1.8'
    targetCompatibility = '1.8'
    // need to run only on >= 10
    options.release.set(8)
}
```
* Using any version JDK supporting gradle to run gradle and using java toolchain in gradle like this PR current commit.
```
java {
    toolchain {
        languageVersion = JavaLanguageVersion.of(8)
    }
}

def javaVersions = [11, 17]
javaVersions.each { version ->
    def testOnSpecificVersion = tasks.register("testsOn$version", Test) {
        testClassesDirs = testing.suites.test.sources.output.classesDirs
        classpath = testing.suites.test.sources.runtimeClasspath

        javaLauncher = javaToolchains.launcherFor {
            languageVersion = JavaLanguageVersion.of(version)
        }
    }
    tasks.named('check').configure {
        dependsOn(testOnSpecificVersion)
    }
}
```

There should be no big differences between them?

Reply:
@ikhoon 
Does the current setup look good to you?